lib/runtime-core/src/parse.rs

```diff
@@ -292,25 +298,30 @@ pub fn read_module<
                     .map_err(|x| LoadError::Codegen(format!("{:?}", x)))?;
                 func_count = func_count.wrapping_add(1);
             }
-            ParserState::BeginActiveElementSectionEntry(table_index) => {
-                let table_index = TableIndex::new(table_index as usize);
+            ParserState::BeginElementSectionEntry {
+                table: ElemSectionEntryTable::Active(table_index_raw),
+                ty: WpType::Func,
```

Comment:
I think you meant this:
```suggestion
                ty: WpType::AnyFunc,
```